tests/tt_eager/python_api_testing/unit_testing/misc/test_distributed_layernorm.py

```diff
@@ -0,0 +1,223 @@
+                dtype=ttnn.bfloat16,
+                device=devices[d],
+                layout=ttnn.ROW_MAJOR_LAYOUT,
+                memory_config=ttnn.DRAM_MEMORY_CONFIG,
+            )
+        )
+
+    tt_beta = []
+    for d in range(n_devices):
+        tt_beta.append(
+            ttnn.as_tensor(
+                beta_chunked[d].reshape(1, 1, -1, 32),
+                dtype=ttnn.bfloat16,
+                device=devices[d],
+                layout=ttnn.ROW_MAJOR_LAYOUT,
+                memory_config=ttnn.DRAM_MEMORY_CONFIG,
+            )
+        )
+    for i in range(iterations):
+        tt_out = tt_distributed_layernorm(
+            tt_inp, tt_gamma, tt_beta, epsilon, is_rmsnorm, compute_kernel_config, stats_dtype
+        )
+        tt_output_host = torch.concat([tt2torch_tensor(tt_o) for tt_o in tt_out], -1)
+
+    passing, output_str = comp_allclose(tt_output_host, out_torch, rtol=1e-1, atol=1e-01)
+    logger.debug(f"torch vs tt distributed layernorm = {output_str}")
+
+    assert passing
+
+
+@pytest.mark.parametrize(
+    "iterations",
+    [1, 11],
+    ids=["loops1", "loops11"],
+)
+@pytest.mark.parametrize(
+    "fp32_enabled",
+    [True, False],
+    ids=["fp32_enabled", "fp32_disabled"],
+)
+@pytest.mark.parametrize(
+    "dtype",
+    (ttnn.bfloat16, ttnn.bfloat8_b),
+    ids=["BFLOAT16_in", "BFLOAT8_B_in"],
+)
+@pytest.mark.parametrize(
+    "stats_dtype",
+    (ttnn.bfloat16, ttnn.bfloat8_b),
+    ids=["BFLOAT16_stats", "BFLOAT8_B_stats"],
+)
+@pytest.mark.parametrize(
+    "inp_shape",
+    [
+        (1, 1, 2048, 8192),
+        (1, 1, 128, 8192),
+        (2, 1, 128, 8192),
+    ],
+)
+@pytest.mark.parametrize(
+    "n_devices",
+    [4, 8],
+)
+@pytest.mark.parametrize(
+    "is_rmsnorm",
+    [True, False],
+    ids=["rmsnorm", "layernorm"],
+)
+def test_distributed_layernorm(
+    inp_shape, n_devices, is_rmsnorm, dtype, stats_dtype, all_devices, fp32_enabled, iterations
+):
+    devices = get_devices_for_t3000(all_devices, n_devices)
+    run_distributed_layernorm(inp_shape, n_devices, is_rmsnorm, dtype, stats_dtype, devices, fp32_enabled, iterations)
+
+
+@pytest.mark.parametrize(
+    "iterations",
+    [1, 11],
+    ids=["loops1", "loops11"],
+)
+@pytest.mark.parametrize(
+    "dtype",
+    (ttnn.bfloat16, ttnn.bfloat8_b),
+    ids=["BFLOAT16_in", "BFLOAT8_B_in"],
+)
+@pytest.mark.parametrize(
+    "stats_dtype",
+    (ttnn.bfloat16, ttnn.bfloat8_b),
+    ids=["BFLOAT16_stats", "BFLOAT8_B_stats"],
+)
+@pytest.mark.parametrize(
+    "inp_shape",
+    [
+        (1, 1, 2048, 8192),
+        (1, 1, 128, 8192),
+        (2, 1, 128, 8192),
+    ],
+)
+@pytest.mark.parametrize(
+    "n_devices",
+    [4, 8],
+)
+@pytest.mark.parametrize(
+    "is_rmsnorm",
+    [True, False],
+    ids=["rmsnorm", "layernorm"],
+)
+def test_distributed_layernorm_with_program_cache(
+    inp_shape, n_devices, is_rmsnorm, dtype, stats_dtype, iterations, all_devices, use_program_cache
```

Comment:
Since this test requires T3000 to run, it needs to skip if len(all_device) < n_devices.  `test_all_gather.py` has an example of how this is done.

Reply:
These `tt_eager` tests should also be added to a t3000 pipeline, probably `run_t3000_frequent_tests.sh`?

Reply:
I added the layernorm test to the t3000 tests. part 1 and part 2 tests run on a single device, what's the proper pipeline for those?

Reply:
Because part1 and part2 tests are in that folder, they should get executed tt_eager tests in pipelines automatically afaik